numpy/random/tests/test_random.py

```diff
@@ -639,6 +639,77 @@ def test_multivariate_normal(self):
         cov = [[1, 1 + 1e-10], [1 + 1e-10, 1]]
         assert_warns(RuntimeWarning, np.random.multivariate_normal, mean, cov)
 
+    def test_multivariate_hypergeometric(self):
+        np.random.seed(self.seed)
+
+        # Error cases...
+        # `colors` must be a 1-d sequence
+        assert_raises(ValueError, np.random.multivariate_hypergeometric,
+                      10, 4)
+        # Negative nsample
+        assert_raises(ValueError, np.random.multivariate_hypergeometric,
+                      [2, 3, 4], -1)
+        # Negative color
+        assert_raises(ValueError, np.random.multivariate_hypergeometric,
+                      [-1, 2, 3], 2)
+        # nsample exceeds sum(colors)
+        assert_raises(ValueError, np.random.multivariate_hypergeometric,
+                      [2, 3, 4], 10)
+        # nsample exceeds sum(colors) (edge case of empty colors)
+        assert_raises(ValueError, np.random.multivariate_hypergeometric,
+                      [], 1)
+
+        # Edge cases...
+        sample = np.random.multivariate_hypergeometric([0, 0, 0], 0)
+        assert_array_equal(sample, [0, 0, 0])
+        sample = np.random.multivariate_hypergeometric([], 0)
+        assert_array_equal(sample, [])
+        sample = np.random.multivariate_hypergeometric([1, 2, 3], 0)
+        assert_array_equal(sample, [0, 0, 0])
+        sample = np.random.multivariate_hypergeometric([9, 0, 0], 3)
+        assert_array_equal(sample, [3, 0, 0])
+        sample = np.random.multivariate_hypergeometric([1, 1, 0, 1, 1], 4)
+        assert_array_equal(sample, [1, 1, 0, 1, 1])
+
+        # "Typical" cases...
+        colors = np.array([10, 5, 20])
+        nsample = 8
+
+        size = 5
+        sample = np.random.multivariate_hypergeometric(colors, nsample, size)
+        assert_array_equal(sample.sum(axis=1),
+                           np.full(size, fill_value=nsample, dtype=int))
+        assert_((sample <= colors).all())
+
+        size = (2, 3)
+        sample = np.random.multivariate_hypergeometric(colors, nsample, size)
+        assert_array_equal(sample.sum(axis=-1),
+                           np.full(size, fill_value=nsample, dtype=int))
```

Comment:
I noticed that, but I didn't use it because it is not actually documented in the function's docstring, and I don't think it should behave that way.  If I expect a result to be [10, 10, 10], and I use `assert_array_equal(result, 10)`, it will (incorrectly) pass if `result` is 10.  Note that using a scalar value to check that an array contains all those values actually contradicts the docstring, which says "...check that the shape is equal...".  So I would say the fact that `assert_array_equal([10, 10, 10], 10)` doesn't raise an exception is a bug.